inductor: Don't throw an internal error when a nn.module is missing a attribute

c00w opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #145122

If a nn.module getattr call throws, we should make sure that we don't crash with an internal error

Note that I couldn't figure out how to test this, so advice would be awesome.  I have my best case attempt at  https://github.com/pytorch/pytorch/pull/145799, but it doesn't seem to reproduce the crash.

cc @voznesenskym @penguinwu @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @wenzhe-nrv @jiayisunx @chenyang78 @kadeng @chauhang @amjames

c00w committed:
inductor: Don't throw an internal error when a nn.module is missing a attribute

If a nn.module getattr call throws, we should make sure that we don't crash with an internal error

[ghstack-poisoned]

c00w committed:
Update on "inductor: Don't throw an internal error when a nn.module is missing a attribute"


If a nn.module getattr call throws, we should make sure that we don't crash with an internal error

cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng wenzhe-nrv jiayisunx chenyang78 kadeng chauhang amjames

[ghstack-poisoned]

c00w committed:
Update on "inductor: Don't throw an internal error when a nn.module is missing a attribute"


If a nn.module getattr call throws, we should make sure that we don't crash with an internal error

cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng wenzhe-nrv jiayisunx chenyang78 kadeng chauhang amjames

[ghstack-poisoned]

c00w committed:
None on "inductor: Don't throw an internal error when a nn.module is missing a attribute"


If a nn.module getattr call throws, we should make sure that we don't crash with an internal error

cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng wenzhe-nrv jiayisunx chenyang78 kadeng chauhang amjames

[ghstack-poisoned]

c00w committed:
Update on "inductor: Don't throw an internal error when a nn.module is missing a attribute"


If a nn.module getattr call throws, we should make sure that we don't crash with an internal error

cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng wenzhe-nrv jiayisunx chenyang78 kadeng chauhang amjames

[ghstack-poisoned]

c00w committed:
Update on "inductor: Don't throw an internal error when a nn.module is missing a attribute"


If a nn.module getattr call throws, we should make sure that we don't crash with an internal error

Note that I couldn't figure out how to test this, so advice would be awesome.  I have my best case attempt at  https://github.com/pytorch/pytorch/pull/145799, but it doesn't seem to reproduce the crash.

cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng wenzhe-nrv jiayisunx chenyang78 kadeng chauhang amjames

[ghstack-poisoned]

c00w committed:
Update on "inductor: Don't throw an internal error when a nn.module is missing a attribute"


If a nn.module getattr call throws, we should make sure that we don't crash with an internal error

Note that I couldn't figure out how to test this, so advice would be awesome.  I have my best case attempt at  https://github.com/pytorch/pytorch/pull/145799, but it doesn't seem to reproduce the crash.

cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng wenzhe-nrv jiayisunx chenyang78 kadeng chauhang amjames

[ghstack-poisoned]

jansel commented on torch/_dynamo/variables/nn_module.py:
```diff
@@ -248,6 +248,7 @@ def _custom_getattr_fallback(self, base, tx, name, obj_source):
         )
 
     def var_getattr(self, tx: "InstructionTranslator", name):
+        print("VAR_GETAATTR")
```
remove

jansel commented on torch/_dynamo/variables/nn_module.py:
```diff
@@ -286,8 +287,9 @@ def var_getattr(self, tx: "InstructionTranslator", name):
-                # if we can't find a __getattr__, just raise the AttributeError
-                raise
+                # if we can't find a __getattr__, we can't parse this, raise unimplemented
+                print("HIT IT")
```
remove

c00w replied:
Fixed

jansel reviewed (changes requested):
Can you add a test for this?

c00w committed:
Update on "inductor: Don't throw an internal error when a nn.module is missing a attribute"


If a nn.module getattr call throws, we should make sure that we don't crash with an internal error

Note that I couldn't figure out how to test this, so advice would be awesome.  I have my best case attempt at  https://github.com/pytorch/pytorch/pull/145799, but it doesn't seem to reproduce the crash.

cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng wenzhe-nrv jiayisunx chenyang78 kadeng chauhang amjames

[ghstack-poisoned]

c00w committed:
Update on "inductor: Don't throw an internal error when a nn.module is missing a attribute"


If a nn.module getattr call throws, we should make sure that we don't crash with an internal error

Note that I couldn't figure out how to test this, so advice would be awesome.  I have my best case attempt at  https://github.com/pytorch/pytorch/pull/145799, but it doesn't seem to reproduce the crash.

cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng wenzhe-nrv jiayisunx chenyang78 kadeng chauhang amjames

[ghstack-poisoned]

c00w committed:
Update on "inductor: Don't throw an internal error when a nn.module is missing a attribute"


If a nn.module getattr call throws, we should make sure that we don't crash with an internal error

Note that I couldn't figure out how to test this, so advice would be awesome.  I have my best case attempt at  https://github.com/pytorch/pytorch/pull/145799, but it doesn't seem to reproduce the crash.

cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng wenzhe-nrv jiayisunx chenyang78 kadeng chauhang amjames

[ghstack-poisoned]

c00w commented:
@jansel Test added, had to turn off inline_inbuilt_nn_modules to get this built.

c00w committed:
Update on "inductor: Don't throw an internal error when a nn.module is missing a attribute"


If a nn.module getattr call throws, we should make sure that we don't crash with an internal error

Note that I couldn't figure out how to test this, so advice would be awesome.  I have my best case attempt at  https://github.com/pytorch/pytorch/pull/145799, but it doesn't seem to reproduce the crash.

cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng wenzhe-nrv jiayisunx chenyang78 kadeng chauhang amjames

[ghstack-poisoned]

c00w commented:
@pytorchbot merge

c00w commented:
@pytorchbot merge -f "unrelated failure afaict - around version satisfaction which I didn't touch"

ZainRizvi commented:
@pytorchbot revert -c ghfirst -m "Sorry but this is failing internally. See D68924977 for details"

To repro these failures internally, you can follow the instructions here: https://fburl.com/fixing-ghfirst-reverts

c00w commented:
Update - we depend on attribute error being raised in this code to support certain models of execution. My plan is to raise a UnimplementedAttributeError, so that we can catch this "correctly", while still not breaking compilation if we cannot.

c00w committed:
Update on "inductor: Don't throw an internal error when a nn.module is missing a attribute"


If a nn.module getattr call throws, we should make sure that we don't crash with an internal error

Note that I couldn't figure out how to test this, so advice would be awesome.  I have my best case attempt at  https://github.com/pytorch/pytorch/pull/145799, but it doesn't seem to reproduce the crash.

cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng wenzhe-nrv jiayisunx chenyang78 kadeng chauhang amjames

[ghstack-poisoned]

c00w committed:
Update on "inductor: Don't throw an internal error when a nn.module is missing a attribute"


If a nn.module getattr call throws, we should make sure that we don't crash with an internal error

Note that I couldn't figure out how to test this, so advice would be awesome.  I have my best case attempt at  https://github.com/pytorch/pytorch/pull/145799, but it doesn't seem to reproduce the crash.

cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng wenzhe-nrv jiayisunx chenyang78 kadeng chauhang amjames

[ghstack-poisoned]

c00w committed:
Update on "inductor: Don't throw an internal error when a nn.module is missing a attribute"


If a nn.module getattr call throws, we should make sure that we don't crash with an internal error

Note that I couldn't figure out how to test this, so advice would be awesome.  I have my best case attempt at  https://github.com/pytorch/pytorch/pull/145799, but it doesn't seem to reproduce the crash.

cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng wenzhe-nrv jiayisunx chenyang78 kadeng chauhang amjames

[ghstack-poisoned]

c00w commented:
Chatted with @anijain2305 switching to raise_observed_exception

c00w committed:
Update on "inductor: Don't throw an internal error when a nn.module is missing a attribute"


If a nn.module getattr call throws, we should make sure that we don't crash with an internal error

Note that I couldn't figure out how to test this, so advice would be awesome.  I have my best case attempt at  https://github.com/pytorch/pytorch/pull/145799, but it doesn't seem to reproduce the crash.

cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng wenzhe-nrv jiayisunx chenyang78 kadeng chauhang amjames

[ghstack-poisoned]

c00w commented:
@pytorchbot merge

c00w committed:
Update on "inductor: Don't throw an internal error when a nn.module is missing a attribute"


If a nn.module getattr call throws, we should make sure that we don't crash with an internal error

Note that I couldn't figure out how to test this, so advice would be awesome.  I have my best case attempt at  https://github.com/pytorch/pytorch/pull/145799, but it doesn't seem to reproduce the crash.

cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng wenzhe-nrv jiayisunx chenyang78 kadeng chauhang amjames

[ghstack-poisoned]

c00w commented:
@pytorchbot merge
